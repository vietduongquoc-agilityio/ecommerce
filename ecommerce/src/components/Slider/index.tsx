"use client";

import { SetStateAction, useState } from "react";
import { colors, fonts } from "@/themes";
import Button from "../Button";
import Image from "next/image";
import product1 from "@/assets/Images/Product1.png";
import product2 from "@/assets/Images/Product2.png";
import product3 from "@/assets/Images/Product3.png";
import product4 from "@/assets/Images/Product4.png";
import arowSlider from "@/assets/Images/Arow-slider.png";

const images = [product1, product2, product3, product4];

const Slider = () => {
  const [slideIndex, setSlideIndex] = useState(0);

  const nextSlide = () => {
    setSlideIndex((prev) => (prev + 1) % images.length);
  };

  const goToSlide = (index: SetStateAction<number>) => {
    setSlideIndex(index);
  };

  return (
    <section
      style={{
        width: "100%",
        height: "auto",
        background: colors.primary.light,
        display: "flex",
        alignItems: "center",
        justifyContent: "space-between",
        padding: "50px 100px",
        position: "relative",
      }}
    >
      <aside style={{ maxWidth: "500px" }}>
        <h2
          style={{
            fontSize: fonts.size.xxl,
            fontWeight: fonts.weight.bold,
            color: colors.secondary.black,
            marginBottom: "10px",
          }}
        >
          50+ Beautiful rooms inspiration
        </h2>
        <p
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.medium,
            color: colors.secondary.granite,
            marginBottom: "25px",
          }}
        >
          Our designer already made a lot of beautiful prototype rooms that
          inspire you.
        </p>
        <Button variant="primaryXs" size="md">
          Explore More
        </Button>
      </aside>

      {/* Image Slider */}
      <aside style={{ position: "relative", width: "400px" }}>
        <Image
          src={images[slideIndex]}
          alt={`slide-${slideIndex}`}
          width={400}
          style={{
            width: "100%",
            height: "auto",
            transition: "opacity 0.5s ease-in-out",
          }}
        />

        <Image
          src={arowSlider}
          alt={"arowSlider"}
          onClick={nextSlide}
          style={{
            position: "absolute",
            top: "50%",
            right: "-35px",
            transform: "translateY(-50%)",
            border: "none",
            cursor: "pointer",
          }}
        />

        {/* Dots Indicators */}
        <div
          style={{
            position: "absolute",
            bottom: "10px",
            left: "50%",
            transform: "translateX(-50%)",
            display: "flex",
            gap: "5px",
          }}
        >
          {images.map((_, index) => (
            <span
              key={index}
              onClick={() => goToSlide(index)}
              style={{
                width: "10px",
                height: "10px",
                borderRadius: "50%",
                background:
                  index === slideIndex ? colors.primary.yellow : "white",
                cursor: "pointer",
              }}
            />
          ))}
        </div>
      </aside>
    </section>
  );
};

export default Slider;
