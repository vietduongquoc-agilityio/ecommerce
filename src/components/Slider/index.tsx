"use client";

import { SetStateAction, useState } from "react";
import { colors, fonts } from "@/themes";
import Image from "next/image";

// Images
import product1 from "@/assets/Images/Slider.jpg";
import product3 from "@/assets/Images/Slider1.jpg";
import product2 from "@/assets/Images/Slider2.jpg";
import product4 from "@/assets/Images/Slider3.webp";
import arowSlider from "@/assets/Images/Arow-slider.png";
import CTA from "@/assets/Images/CTA.png";
import { Button } from "..";

// Components


const images = [product1, product2, product3, product4];
const roomNames = [
  "01 - Living Room",
  "02 - Bed Room",
  "03 - Dining Room",
  "04 - Office Room",
];

const Slider = () => {
  const [slideIndex, setSlideIndex] = useState(0);

  const nextSlide = () => {
    setSlideIndex((prev) => (prev + 1) % images.length);
  };

  const goToSlide = (index: SetStateAction<number>) => {
    setSlideIndex(index);
  };

  return (
    <section
      style={{
        width: "100%",
        height: "auto",
        background: colors.primary.light,
        display: "flex",
        alignItems: "center",
        justifyContent: "space-between",
        padding: "50px 100px",
        position: "relative",
      }}
    >
      <aside style={{ maxWidth: "500px" }}>
        <h2
          style={{
            fontSize: fonts.size.xxl,
            fontWeight: fonts.weight.bold,
            color: colors.secondary.black,
            marginBottom: "10px",
          }}
        >
          50+ Beautiful rooms inspiration
        </h2>
        <p
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.medium,
            color: colors.secondary.granite,
            marginBottom: "25px",
            width:"368px"
          }}
        >
          Our designer already made a lot of beautiful prototype rooms that
          inspire you.
        </p>
        <Button variant="primaryXs" size="md">
          Explore More
        </Button>
      </aside>

      {/* Image Slider */}
      <aside
        style={{ position: "relative", width: "650px", overflow: "hidden" }}
      >
        <div
          style={{
            display: "flex",
            transition: "transform 0.5s ease-in-out",
            transform: `translateX(-${slideIndex * 100}%)`,
          }}
        >
          {images.map((img, index) => (
            <div
              key={index}
              style={{
                minWidth: "70%",
                position: "relative",
                display: "flex",
              }}
            >
              <Image
                src={img}
                alt={`slide-${index}`}
                style={{
                  width: "400px",
                  height: "582px",
                  transition: "opacity 0.5s ease-in-out",
                }}
              />
              <div
                style={{
                  position: "absolute",
                  bottom: "30px",
                  left: "24px",
                  background: "#FFFFFFB8",
                  padding: "30px 17px 32px 32px",
                }}
              >
                <p
                  style={{
                    fontSize: fonts.size.xs,
                    fontWeight: fonts.weight.medium,
                    color: colors.secondary.granite,
                  }}
                >
                  {roomNames[index]}
                </p>
                <p
                  style={{
                    fontSize: "28px",
                    fontWeight: fonts.weight.semiBold,
                    color: colors.secondary.black,
                  }}
                >
                  Inner Peace
                </p>
                <Image
                  src={CTA}
                  alt={"CTA"}
                  style={{ right: "-48px", position: "absolute", bottom: "0" }}
                />
              </div>
            </div>
          ))}
        </div>

        {/* Next Slide Button */}
        <Image
          src={arowSlider}
          alt={"arowSlider"}
          onClick={nextSlide}
          style={{
            position: "absolute",
            top: "50%",
            right: "0",
            transform: "translateY(-50%)",
            border: "none",
            cursor: "pointer",
            width: "60px",
            height: "60px",
          }}
        />

        {/* Dots Indicators */}
        <div
          style={{
            position: "absolute",
            bottom: "10px",
            left: "50%",
            transform: "translateX(-50%)",
            display: "flex",
            gap: "5px",
          }}
        >
          {images.map((_, index) => (
            <span
              key={index}
              onClick={() => goToSlide(index)}
              style={{
                width: "10px",
                height: "10px",
                borderRadius: "50%",
                background:
                  index === slideIndex ? colors.primary.yellow : "white",
                cursor: "pointer",
              }}
            />
          ))}
        </div>
      </aside>
    </section>
  );
};

export default Slider;
