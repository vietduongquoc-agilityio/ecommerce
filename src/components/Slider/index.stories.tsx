import type { Meta, StoryObj } from "@storybook/react";
import Slider from "./";

const meta: Meta<typeof Slider> = {
  title: "Components/Slider",
  component: Slider,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof Slider>;

export const Default: Story = {};
