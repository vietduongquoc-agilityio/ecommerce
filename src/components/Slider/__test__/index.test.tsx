import { render } from "@testing-library/react";
import Slider from "../index";
import { useRouter } from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

const mockPush = jest.fn();
describe("Slider Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();

    (useRouter as jest.Mock).mockReturnValue({ push: mockPush });
  });
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Slider />);
    expect(asFragment()).toMatchSnapshot();
  });
});
