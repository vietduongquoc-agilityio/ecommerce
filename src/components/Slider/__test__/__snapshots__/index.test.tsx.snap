// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component matches snapshot 1`] = `
<DocumentFragment>
  <section
    style="width: 100%; height: auto; background: rgb(255, 243, 227); padding: 50px 0px; position: relative; margin-top: 150px; margin-bottom: 67px;"
  >
    <div
      style="display: flex; align-items: center; justify-content: space-between; width: 1240px; margin: 0px auto;"
    >
      <aside
        style="max-width: 500px;"
      >
        <h2
          style="font-size: 40px; font-weight: 700; color: rgb(58, 58, 58); margin-bottom: 10px;"
        >
          50+ Beautiful rooms inspiration
        </h2>
        <p
          style="font-size: 16px; font-weight: 500; color: rgb(102, 102, 102); margin-bottom: 25px; width: 368px;"
        >
          Our designer already made a lot of beautiful prototype rooms that inspire you.
        </p>
        <button
          style="padding: 12px 36px; font-size: 16px; font-weight: 600; text-align: center; width: auto; height: auto; cursor: pointer; transition: all 0.3s ease; background-color: rgb(184, 142, 47); color: rgb(255, 255, 255);"
        >
          Explore More
        </button>
      </aside>
      <aside
        style="position: relative; width: 650px; overflow: hidden;"
      >
        <div
          style="display: flex; transition: transform 0.5s ease-in-out; transform: translateX(-0%);"
        >
          <div
            style="min-width: 70%; position: relative; display: flex;"
          >
            <img
              alt="slide-0"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent; width: 400px; height: 582px; transition: opacity 0.5s ease-in-out;"
              width="40"
            />
            <div
              style="position: absolute; bottom: 30px; left: 24px; background: rgba(255, 255, 255, 0.722); padding: 30px 17px 32px 32px;"
            >
              <p
                style="font-size: 16px; font-weight: 500; color: rgb(102, 102, 102);"
              >
                01 - Living Room
              </p>
              <p
                style="font-size: 28px; font-weight: 600; color: rgb(58, 58, 58);"
              >
                Inner Peace
              </p>
              <img
                alt="CTA"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent; right: -48px; position: absolute; bottom: 0px;"
                width="40"
              />
            </div>
          </div>
          <div
            style="min-width: 70%; position: relative; display: flex;"
          >
            <img
              alt="slide-1"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent; width: 400px; height: 582px; transition: opacity 0.5s ease-in-out;"
              width="40"
            />
            <div
              style="position: absolute; bottom: 30px; left: 24px; background: rgba(255, 255, 255, 0.722); padding: 30px 17px 32px 32px;"
            >
              <p
                style="font-size: 16px; font-weight: 500; color: rgb(102, 102, 102);"
              >
                02 - Bed Room
              </p>
              <p
                style="font-size: 28px; font-weight: 600; color: rgb(58, 58, 58);"
              >
                Inner Peace
              </p>
              <img
                alt="CTA"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent; right: -48px; position: absolute; bottom: 0px;"
                width="40"
              />
            </div>
          </div>
          <div
            style="min-width: 70%; position: relative; display: flex;"
          >
            <img
              alt="slide-2"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent; width: 400px; height: 582px; transition: opacity 0.5s ease-in-out;"
              width="40"
            />
            <div
              style="position: absolute; bottom: 30px; left: 24px; background: rgba(255, 255, 255, 0.722); padding: 30px 17px 32px 32px;"
            >
              <p
                style="font-size: 16px; font-weight: 500; color: rgb(102, 102, 102);"
              >
                03 - Dining Room
              </p>
              <p
                style="font-size: 28px; font-weight: 600; color: rgb(58, 58, 58);"
              >
                Inner Peace
              </p>
              <img
                alt="CTA"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent; right: -48px; position: absolute; bottom: 0px;"
                width="40"
              />
            </div>
          </div>
          <div
            style="min-width: 70%; position: relative; display: flex;"
          >
            <img
              alt="slide-3"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent; width: 400px; height: 582px; transition: opacity 0.5s ease-in-out;"
              width="40"
            />
            <div
              style="position: absolute; bottom: 30px; left: 24px; background: rgba(255, 255, 255, 0.722); padding: 30px 17px 32px 32px;"
            >
              <p
                style="font-size: 16px; font-weight: 500; color: rgb(102, 102, 102);"
              >
                04 - Office Room
              </p>
              <p
                style="font-size: 28px; font-weight: 600; color: rgb(58, 58, 58);"
              >
                Inner Peace
              </p>
              <img
                alt="CTA"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent; right: -48px; position: absolute; bottom: 0px;"
                width="40"
              />
            </div>
          </div>
        </div>
        <img
          alt="arowSlider"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent; position: absolute; top: 50%; right: 0px; transform: translateY(-50%); cursor: pointer; width: 60px; height: 60px;"
          width="40"
        />
        <div
          style="position: absolute; bottom: 10px; left: 50%; transform: translateX(-50%); display: flex; gap: 5px;"
        >
          <span
            style="width: 10px; height: 10px; border-radius: 50%; background: rgb(184, 142, 47); cursor: pointer;"
          />
          <span
            style="width: 10px; height: 10px; border-radius: 50%; background: white; cursor: pointer;"
          />
          <span
            style="width: 10px; height: 10px; border-radius: 50%; background: white; cursor: pointer;"
          />
          <span
            style="width: 10px; height: 10px; border-radius: 50%; background: white; cursor: pointer;"
          />
        </div>
      </aside>
    </div>
  </section>
</DocumentFragment>
`;
